hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HoodieHiveCatalog.java

```diff
@@ -510,6 +511,9 @@ private void initTableIfNotExists(ObjectPath tablePath, CatalogTable catalogTabl
     }
 
     flinkConf.setString(FlinkOptions.TABLE_NAME, tablePath.getObjectName());
+
+    StreamerUtil.checkPreCombineKey(flinkConf, ((ResolvedCatalogTable) catalogTable).getResolvedSchema());
```

Comment:
<img width="865" alt="image" src="https://github.com/apache/hudi/assets/69956021/cf1e72d6-07ce-4c86-b1a2-3e5a07b556f5">

Looks like UT failed related to this change. please take a look.

Reply:
Also it is better to modify/create related UTs to check percombine field in hoodie.properties